### PR TITLE
Feature location

### PR DESCRIPTION
**Pull request by thinhnguyen91**
New LocationCell
Change background
	Api country
	pop Country -> Location

### CODE REVIEW

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 1]
```diff
@@ -7,12 +7,36 @@
 //
 
 import UIKit
+import ObjectMapper
 
 class HomeViewController: BaseViewController {
-
+    var weather = WebService()
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: missing `private`

File: Weather/Controllers/HomeViewController/HomeViewController.swift [hunk 2]
```diff
@@ -7,12 +7,36 @@
...
+    private var weathers = [Weather]()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        self.weather.getURL { (success, result, error) in
```
Comment (at-phuongvnc) [inline Weather/Controllers/HomeViewController/HomeViewController.swift]: please remove `self`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import ObjectMapper
+
+protocol CountryViewControllerDelegate {
+    func showData(name: String, cell: LocationCell)
+}
+private struct Options {
+    static let heightCell: CGFloat = 50
+}
+class CountryViewController: BaseViewController {
+    var weather = WebService()
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: missing `private`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    var delegate: CountryViewControllerDelegate?
+    var cell: LocationCell?
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: Does it use public ?

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    private var weathers = [Weather]()
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.delegate = self
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: You can use `setUpUI()` or `setUpData`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        let item = weathers[indexPath.row]
+        cell!.nameCountry.text = item.name
+
+        return cell!
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        let cellrow = tableView.cellForRowAtIndexPath(indexPath) as? CountryCell
+        let delegate = self.delegate
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: what is this ?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 7]
```diff
@@ -7,50 +7,75 @@
...
     @IBOutlet private weak var tableView: UITableView!
 
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        self.tableView.reloadData()
+     
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please add it in `setUpUI()` or `setUpData()`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 8]
```diff
@@ -7,50 +7,75 @@
...
+        let cell = self.tableView.dequeueReusableCellWithIdentifier("locationCell") as? LocationCell
+        let item = array[indexPath.row]
+        cell?.nameLabel.text = item
 
         return cell!
     }
     func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
         self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
-
+        let index = indexPath.row
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: Why do you use it ?

**Comment by at-phuongvnc** [comment]
Please check comment !

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift (hunk 5 above)
Comment (thinhnguyen91) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: finished

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift (hunk 6 above)
Comment (thinhnguyen91) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: finished

File: Weather/Controllers/LocationViewController/LocationViewController.swift (hunk 7 above)
Comment (thinhnguyen91) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: finished

File: Weather/Controllers/LocationViewController/LocationViewController.swift (hunk 8 above)
Comment (thinhnguyen91) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: finished

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    }
+    // MARK: Action
+
+    @IBAction private func backAction(sender: AnyObject) {
+
+        delegate?.show(isCheck)
+        navigationController?.popViewControllerAnimated(true)
+    }
+
+    func getJson() {
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: missing `private`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        return Options.heightCell
+    }
+
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return weathers.count
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = self.tableView.dequeueReusableCellWithIdentifier("countryCell") as? CountryCell
+        let item = weathers[indexPath.row]
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: please check it before you use it

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+        cell!.nameCountry.text = item.name
+
+        return cell!
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        let cellrow = tableView.cellForRowAtIndexPath(indexPath) as? CountryCell
+        let item = weathers[indexPath.row]
+        print(item.altSpelling)
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: remove `print`

File: Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+        let item = weathers[indexPath.row]
+        cell!.nameCountry.text = item.name
+
+        return cell!
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        let cellrow = tableView.cellForRowAtIndexPath(indexPath) as? CountryCell
+        let item = weathers[indexPath.row]
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/CountryViewController/CountryViewController.swift]: please check it

File: Weather/Controllers/LocationViewController/LocationCell/LocationCell.swift [hunk 13]
```diff
@@ -8,17 +8,19 @@
 
 import UIKit
 
-class CountryTableViewCell: BaseTableViewCell {
+class LocationCell: BaseTableViewCell {
+
+    @IBOutlet weak var nameLabel: UILabel!
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationCell/LocationCell.swift]: missing `private`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 14]
```diff
@@ -7,50 +7,128 @@
...
+    var country = ""
+    private var indexpath = NSIndexPath()
     @IBOutlet private weak var tableView: UITableView!
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        self.tableView.reloadData()
+        print(city)
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: remove `print`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 15]
```diff
@@ -7,50 +7,128 @@
...
+    override func setUpData() {
+        tableView.tableHeaderView = UIView(frame: CGRect(x: 0, y: 0, width: 0.01, height: 0.01))
+        let nib = UINib(nibName: "LocationCell", bundle: nil)
+        tableView.registerNib(nib, forCellReuseIdentifier: "locationCell")
+    }
     // MARK: - Action
+    @IBAction func saveAction(sender: AnyObject) {
+
+        let myRegister = RegisterViewController()
+        myRegister.cell = tableView.cellForRowAtIndexPath(indexpath) as? LocationCell
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: wtf?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 16]
```diff
@@ -7,50 +7,128 @@
...
 
     }
     func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 2
+        return array.count
     }
     func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = self.tableView.dequeueReusableCellWithIdentifier("CountryCell") as? CountryTableViewCell
-
+        let cell = self.tableView.dequeueReusableCellWithIdentifier("locationCell") as? LocationCell
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: remove `self`

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 17]
```diff
@@ -7,50 +7,128 @@
...
+        let item = array[indexPath.row]
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: please check it

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 18]
```diff
@@ -7,50 +7,128 @@
...
+        }
         return cell!
     }
     func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
         self.tableView.deselectRowAtIndexPath(indexPath, animated: true)
+        let index = indexPath.row
+        if index == 0 {
+            let myCountryVC = CountryViewController()
+            myCountryVC.delegate = self
+            myCountryVC.cell = tableView.cellForRowAtIndexPath(indexPath) as? LocationCell
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: wtf ?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 19]
```diff
@@ -7,50 +7,128 @@
...
+            isCheckCountry = true
+            self.navigationController?.pushViewController(myCountryVC, animated: true)
+
+        }
+        if isCheckCountry {
+            if index == 1 {
+
+                let myCityVC = CityViewController()
+                myCityVC.delegate = self
+                myCityVC.cell = tableView.cellForRowAtIndexPath(indexPath) as? LocationCell
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: wtf ?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 20]
```diff
@@ -7,50 +7,128 @@
...
     func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
 
     }
-}
\ No newline at end of file
+}
+extension LocationViewController: CountryViewControllerDelegate {
+    func showData(name: String, cell: LocationCell) {
+        cell.titleLabel.text = name
+        self.country = cell.titleLabel.text!
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: wtf ?

File: Weather/Controllers/LocationViewController/LocationViewController.swift [hunk 21]
```diff
@@ -7,50 +7,128 @@
...
+
+    func showSpell(altspell: String) {
+        self.altspell = altspell
+    }
+
+}
+extension LocationViewController: CityViewControllerDelegate {
+    func showDataCity(name: String, cell: LocationCell) {
+        cell.titleLabel.text = name
+        self.city = cell.titleLabel.text!
```
Comment (at-phuongvnc) [inline Weather/Controllers/LocationViewController/LocationViewController.swift]: wtf ?

File: Weather/Controllers/RegisterViewController/RegisterViewController.swift [hunk 22]
```diff
@@ -7,8 +7,18 @@
 //
 
 import UIKit
-
+import ObjectMapper
+protocol RegisterViewControllerDelegate {
+    func pushData (name: String, cell: LocationCell)
```
Comment (at-phuongvnc) [inline Weather/Controllers/RegisterViewController/RegisterViewController.swift]: wtf ?

File: Weather/Controllers/LocationViewController/LocationCell/LocationCell.swift (hunk 13 above)
Comment (thinhnguyen91) [inline Weather/Controllers/LocationViewController/LocationCell/LocationCell.swift]: screen shot 2016-07-12 at 10 58 28
private error

**Comment by at-phuongvnc** [comment]
LGTM 👯

**Pull request merged**